src/components/todoListItem.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from 'react';
+
+const TodoListItem = ({ title, content, onRemove, onUpdate,isEdit,onMoveProgress,onMoveDone,onMoveTodo }) => {
+  const [isEditing, setIsEditing] = useState(isEdit);
+  const [updatedTitle, setUpdatedTitle] = useState(title);
+  const [updatedContent, setUpdatedContent] = useState(content);
+  const [isHovered, setIsHovered] = useState(false);
+
+  const handleRemove = () => {
+    onRemove();
+  };
+
+  const handleUpdate = () => {
+    onUpdate(updatedTitle, updatedContent);
+    setIsEditing(false);
+  };
+  const handleMoveProgress=()=>{
+    onMoveProgress();
+  }
+  const handleMoveDone=()=>{
+    onMoveDone();
+  }
+  const handleMoveTodo=()=>{
+    onMoveTodo();
+  }
+  return (
+    <div className="todoListItem" 
+    onMouseEnter={() => setIsHovered(true)}
+    onMouseLeave={() => setIsHovered(false)}>
+      {isHovered && !isEditing && (
+        <div className="hoverButtons">
+          <button className="hoverMoveTodoBtn" onClick={handleMoveTodo}>Move To do 🐣</button>
+          <button className="hoverMoveProgressBtn" onClick={handleMoveProgress}>Move Progress 🐥</button>
+          <button className="hoverMoveDoneBtn" onClick={handleMoveDone}>Move Done 🕊</button>
+        </div>
+      )}
+      {isEditing ? (
+        // 수정 모드일 때
+        <div className="todoListEditContainer">
+          <input
+            className="todoListEditTitle"
+            type="text"
+            value={updatedTitle}
+            onChange={(e) => setUpdatedTitle(e.target.value)}
+          />
```

Comment:
일반 모드일 때 readonly나 disabled속성을 줘서 컴포넌트를 굳이 만들지 않고 사용할 수도 있을것같습니다
```suggestion
          <input
            className="todoListEditTitle"
            type="text"
            value={updatedTitle}
            onChange={(e) => setUpdatedTitle(e.target.value)}
            readonly={true}
          />
```